sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -335,14 +335,18 @@ class CudfSum(override val dataType: DataType) extends CudfAggregate with Arm {
   override val name: String = "CudfSum"
 }
 
-class CudfMax(override val dataType: DataType) extends CudfAggregate {
+class CudfMax(override val dataType: DataType) extends CudfAggregate with Arm {
   override lazy val reductionAggregate: cudf.ColumnVector => cudf.Scalar =
     (col: cudf.ColumnVector) => col.max
   override lazy val groupByAggregate: GroupByAggregation =
     GroupByAggregation.max()
   override val name: String = "CudfMax"
 }
 
+object CudfAny {
+  def apply(): CudfAggregate = new CudfMax(BooleanType)
+}
```

Comment:
Why an object `CudfAny` applies `CudfMax`?

Reply:
If this object is used for some special purpose, please document it and give it a more expressive name.

Reply:
Based on @revans2 's comment https://github.com/NVIDIA/spark-rapids/pull/5989#discussion_r926758210
I will add more docs.

Reply:
Updated.

Reply:
I think a little more documentation would be good. Just saying something like.

```
The CUDF any aggregation does not work for reductions or group by aggregations so we use Max as a workaround for this.
```

Reply:
Updated.